sys/parsoid.js

```diff
@@ -236,72 +394,47 @@ class ParsoidService {
-
-            if (parentRev) {
-                // OnEdit job update: pass along the predecessor version
-                parsoidReq = getOrigAndPostToParsoid(pageBundleUri, `${parentRev}`, 'previous');
-            } else if (updateMode) {
-                // Template or image updates. Similar to html2wt, pass:
-                // - current data-parsoid and html
-                // - the edit mode
-                parsoidReq = getOrigAndPostToParsoid(pageBundleUri, rp.revision,
-                        'original', updateMode);
-            } else {
-                // Plain render
-                parsoidReq = hyper.get({ uri: pageBundleUri });
-            } */
 
             return P.join(parsoidReq, mwUtil.decodeBody(currentContentRes))
             .spread((res, currentContentRes) => {
                 const tid  = uuid.now().toString();
+                const etag = mwUtil.makeETag(rp.revision, tid);
                 res.body.html.body = insertTidMeta(res.body.html.body, tid);
+                res.body.html.headers.etag = res.headers.etag = etag;
 
-                if (format === 'html' &&
-                        currentContentRes &&
+                if (currentContentRes &&
                         currentContentRes.status === 200 &&
-                        sameHtml(res.body.html.body, currentContentRes.body) &&
-                        currentContentRes.headers['content-type'] ===
+                        sameHtml(res.body.html.body, currentContentRes.body.html.body) &&
+                        currentContentRes.body.html.headers['content-type'] ===
                                 res.body.html.headers['content-type']) {
                     // New render is the same as the previous one, no need to store it.
                     hyper.metrics.increment('sys_parsoid_generateAndSave.unchanged_rev_render');
                     return currentContentRes;
                 } else if (res.status === 200) {
-                    const resp = {
-                        status: res.status,
-                        headers: res.body[format].headers,
-                        body: res.body[format].body
-                    };
-                    resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-                    let newContent = true;
-                    return this.saveParsoidResultToLatest(hyper, req, tid, res)
-                    .catch({ status: 412 }, () => {
-                        newContent = false;
-                        // TODO: This will only need to be happening
-                        //  if we're requested by VE with a special flag
-                        return this.saveParsoidResultToFallback(hyper, req, tid, res);
-                    })
-                    .then(() => {
+                    let newContent = false;
+                    return this.saveParsoidResultToLatest(hyper, rp.domain, rp.title, res)
+                    .then((saveRes) => {
+                        if (saveRes.status === 201) {
```

Comment:
If it's not the latest, wouldn't the previous promise throw? I.e. how can we get here if the content is not the latest?

Reply:
If it's not the latest, saveRes will be the previous latest and the status will be 200. If it was the latest and we saved - status will be 201.